rsplan/primitives.py

```diff
@@ -307,9 +309,8 @@ def _interpolated(
         # straight segments) and dtheta (step size / turn radius) for curved segments.
         step = step_size if self.is_straight else step_size / self.turn_radius
 
-        seg_pts = np.arange(0, magnitude, step)
-
-        # Add segment endpoint if the list of segment points is not empty
-        seg_pts = np.append(seg_pts, [magnitude]) if seg_pts.any() else np.array([0.0])
 
-        return seg_pts
+        # Prefer linspace to arange to avoid floating point errors. Will distribute
+        # remainder distance across steps instead of having a short final step.
+        num_steps = math.ceil(magnitude / step)
+        return np.linspace(0, magnitude, num_steps, endpoint=True)
```

Comment:
I think the use of arange vs linspace is mainly for speed. can you verify overall performance difference between versions and see the impact

```python
In [28]: %timeit np.arange(1, 1.5, 0.1)
517 ns ± 1.56 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [29]: %timeit np.linspace(1, 1.5, 6, endpoint=True)
22.6 µs ± 259 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)

```

Reply:
I was shocked to find that using `%timeit` I was getting within a micro second of run time of between the `linspace` and `arange` implementations. The test was this + swapping out the interpolating code:
```
def generate_path_waypoints():
    start_pt = (0,0,0)
    end_pt = (0,5,0)
    path = planner.path(
        start_pt,
        end_pt,
        _TURN_RADII[0],
        _RUNWAY_LENGTHS[1],
        _STEP_SIZE,
        )
    return path.waypoints()
```
Below are the icicle charts, in the `arange` implementation it is expensive to append the `magnitude`  (upper bound of the arange range) to the end.

Arange:
![image](https://github.com/builtrobotics/rsplan/assets/56242766/f355b3f5-7054-4c6c-8be4-cbe9c2071e97)

Linspace:
![image](https://github.com/builtrobotics/rsplan/assets/56242766/b6ef2e36-6e95-478a-a39f-0be92cb81822)


Since performance is comparable, I think we should use `linspace`. WDYT

Reply:
Interesting! Not sure why timeit is so different. If the overall performance with linspace is similar then I see no issues

Reply:
Maybe just also verify the performance when testing using the new rsplan package version in Mariana and verify the fast system tests pass. If so then I see no reason to not change

Reply:
passed fast system tests